protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArray.java

```diff
@@ -451,6 +466,10 @@ private void applyToNodes(final NodeVisitor action) {
     }
   }
 
+  public Map<Bytes32, Integer> getRootIndices() {
+    return protoArrayIndices.getBlockIndices();
+  }
```

Comment:
I'd love to get to a point where this is not necessary.  We should be encapsulating this info.

Not going to happen in this PR though based on the usages I can see for this (and it's not making things worse as the indices map was always exposed). I think a follow up PR could manage it though which may be worth a bit of time.

Reply:
oh, I see you've already fixed most of them.  I suspect a simple `Function<Bytes32, Integer>` could be passed to `ProtoArrayScoreCalculator` instead of the full map and we'd be rid of it.

Reply:
will raise a new PR, as discussed.